components/Banner/examples/supplemental/MDCBannerView.h

```diff
@@ -18,15 +18,42 @@
 
 __attribute__((objc_subclassing_restricted)) @interface MDCBannerView : UIView
 
-@property(nonatomic, copy, nonnull) NSString *text;
+/**
+ A view that displays the text on a @c MDCBannerView.
+ @discussion Although this property is read-only, its properties are read/write. These properties can be used to configure the text shown on @c MDCBannerView.
+ For example:
+ @code
+ MDCBannerView *bannerView = [[MDCBannerView alloc] init];
+ bannerView.textLabel.font = [UIFont systemFontOfSize: 12];
+ @endcode
+ */
+@property(nonatomic, readonly, strong, nonnull) UILabel *textLabel;
 
-@property(nonatomic, strong, nullable) UIImage *image;  // default is nil
+/**
+ A view that displays the image on a @c MDCBannerView.
+ @discussion Although this property is read-only, its properties are read/write. These properties can be used to configure the image shown on @c MDCBannerView.
+ For example:
+ @code
+ MDCBannerView *bannerView = [[MDCBannerView alloc] init];
+ bannerView.imageView.image = [UIImage imageNamed:@"example-image"];
+ bannerView.imageView.tintColor = UIColor.whiteColor;
+ @endcode
+ If its own property @c image is @c nil, this view won't show on the @c MDCBannerView.
+ */
+@property(nonatomic, readonly, strong, nonnull) UIImageView *imageView;
 
-@property(nonatomic, assign) NSUInteger numberOfButtons;  // default is 1
+/**
+ A leading button that displays on a @c MDCBannerView.
+ @discussion This @c leadingButton displays on the leading edge of the view. If it can't occupy on the same line with @c trailingButton, it always shows above it.
+ The default value is nil.
+ */
+@property(nonatomic, readwrite, strong, nullable) MDCButton *leadingButton;
```

Comment:
Why is this readwrite?

Reply:
This one doesn't have to be `readwrite`. But for the reason listed below, this one is set to match `trailingButton`.

Reply:
Talked offline. Addressed in c17c57d42ab6d485cc6570b2ccb6ded11981d656.